pkg/kubectl/cmd/describe.go

```diff
@@ -100,32 +109,40 @@ func RunDescribe(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []s
 	if err != nil {
 		return err
 	}
-	mapping, err := r.ResourceMapping()
-	if err != nil {
-		return err
-	}
 
-	describer, err := f.Describer(mapping)
-	if err != nil {
-		return err
-	}
+	allErrs := []error{}
 	infos, err := r.Infos()
 	if err != nil {
 		if apierrors.IsNotFound(err) && len(args) == 2 {
+			mapping, err := r.ResourceMapping()
```

Comment:
Moving this line to here blocks our way to run DescribeMatchingResources(), due to this line: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/resource/result.go#L158, so it will always return here.